src/titiler/mosaic/titiler/mosaic/resources/models.py

```diff
@@ -39,7 +38,7 @@ class MosaicEntity(BaseModel):
 rfc3339_regex = re.compile(rfc3339_regex_str)
 
 
-class StacApiQueryRequestBody(Search):
+class StacApiQueryRequestBody(ExtendedSearch):
```

Comment:
stac-pydantic moved the `query` parameter from Search to ExtendedSearch, and now has collections as optional